pkg/schemadsl/compiler/compiler.go

```diff
@@ -94,9 +106,11 @@ func toContextError(errMessage string, node *dslNode, mapper input.PositionMappe
 	}
 
 	return ErrorWithContext{
-		error:       fmt.Errorf("parse error in %s: %s", formattedRange, errMessage),
-		SourceRange: sourceRange,
-		Source:      input.Source(source),
+		error:            fmt.Errorf("parse error in %s: %s", formattedRange, errMessage),
```

Comment:
Isn't the thing you're doing here with `error`/`BaseErrorMessage` basically what you get with golang's error wrapping?

Reply:
Yes, but I wanted it to be explicit from the compiler's error type

Reply:
But whenever you wanted to get the underlying error, you can just `errors.As(err, &BaseError{})` to get it back out. I'm not clear on the advantage to doing it outside the stdlib?

Reply:
Well, we'd have to declare another error type to do that. Do you prefer that approach?

Reply:
Done